corehq/apps/hqwebapp/static/hqwebapp/js/components/rich_text_knockout_bindings.js

```diff
@@ -254,8 +290,89 @@ ko.bindingHandlers.richEditor = {
                 },
             },
             theme: "snow",
+            // Disable Quill's native clipboard and drop handling
+            clipboard: {
+                matchVisual: false,
+            },
         });
 
+        // Handle paste events in the capture phase to intercept before Quill's handlers
+        editor.root.addEventListener('paste', function (e) {
+            if (initialPageData.get("read_only_mode")) {
+                return;
+            }
+
+            if (e.clipboardData && e.clipboardData.items) {
+                const items = e.clipboardData.items;
+
+                for (let i = 0; i < items.length; i++) {
+                    if (items[i].type.indexOf('image') !== -1) {
+                        // Prevent default and stop propagation to avoid Quill's handling
+                        e.preventDefault();
+                        e.stopPropagation();
+
+                        const file = items[i].getAsFile();
+                        if (!file) {
+                            continue;
+                        }
+
+                        uploadAndInsertImage(file, editor);
+                        break; // Only handle the first image
+                    }
+                }
+            }
+        }, true); // Using capture phase to intercept before Quill's handlers
+
+        // Completely disable Quill's drop handler
+        editor.root.addEventListener('drop', function (e) {
+            e.stopPropagation();
+        }, true);
+
+        // Handle drag and drop images
+        editor.root.addEventListener('dragover', function (e) {
+            e.preventDefault();
+            e.stopPropagation();
+            return false;
+        }, { capture: true });
+
+        editor.root.addEventListener('dragleave', function (e) {
+            e.preventDefault();
+            e.stopPropagation();
+            return false;
+        }, { capture: true });
+
+        editor.root.addEventListener('drop', function (e) {
+            // This needs to happen before anything else
```

Comment:
nit: is this just saying that `preventDefault` needs to be called first? If so, nit: not very useful